src/main/java/com/blamejared/contenttweaker/api/CoTRegistry.java

```diff
@@ -1,22 +1,29 @@
 package com.blamejared.contenttweaker.api;
 
-import com.blamejared.contenttweaker.api.blocks.*;
-import com.blamejared.contenttweaker.api.items.*;
-import com.blamejared.contenttweaker.api.resources.*;
-import com.blamejared.crafttweaker.api.*;
-import net.minecraft.block.*;
-import net.minecraft.item.*;
-import net.minecraft.util.*;
-import net.minecraftforge.common.extensions.*;
+import com.blamejared.contenttweaker.ContentTweaker;
+import com.blamejared.contenttweaker.api.blocks.IIsCoTBlock;
+import com.blamejared.contenttweaker.api.fluids.IIsCotFluid;
+import com.blamejared.contenttweaker.api.functions.ICotFunction;
+import com.blamejared.contenttweaker.api.items.IIsCotItem;
+import com.blamejared.contenttweaker.api.resources.WriteableResource;
+import com.blamejared.crafttweaker.api.CraftTweakerAPI;
+import net.minecraft.block.Block;
+import net.minecraft.item.Item;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.common.extensions.IForgeBlock;
+import net.minecraftforge.common.extensions.IForgeItem;
+import org.apache.commons.lang3.tuple.Pair;
 
 import java.util.*;
-import java.util.stream.*;
+import java.util.stream.Stream;
 
 public class CoTRegistry {
     
     private final Map<ResourceLocation, IIsCoTBlock> blocks = new LinkedHashMap<>();
     private final Map<ResourceLocation, IIsCotItem> items = new LinkedHashMap<>();
-    
+    private final Map<ResourceLocation, IIsCotFluid> fluids = new LinkedHashMap<>();
+    private final Map<Pair<ResourceLocation, Class<? extends ICotFunction>>, ICotFunction> functions = new HashMap<>();
```

Comment:
I believe there was a talk about this already, but:
Why is this in here, and not e.g. a part of the classes themselves?

Reply:
To handle all items/blocks' functions, and help me not to add tons of set function action. I think getting elements from HashMap is efficient, too.